### PR TITLE
Add the runtime dependencies of the transitive closure of used ppx rewriters

### PR DESCRIPTION
**Pull request by diml**
This fixes the test added in #449.

I realized that the the function that compute the runtime deps of the transitive closure was trying to be too clever and add scoping issues... I rewrote it

### CODE REVIEW

**Comment by rgrinberg** [comment]
> and add scoping issues

mind sharing a few words about these issues? would like to address this scoping stuff a bit better myself.

**Comment by diml** [comment]
Basically, it is using a `seen` variable to ensure each library is scanned only once. However, for private libraries it was putting directly the name it this, so there could have been an issue. Now it's using a unique identifier.

File: src/lib_db.ml [hunk 1]
```diff
@@ -193,57 +193,59 @@ let best_lib_dep_names_exn t ~dir lib_deps =
...
+        end
+    end
   in
-  List.iter lib_deps ~f:(loop dir);
-  !result
+  List.fold_left lib_deps ~init ~f:(loop dir)
+
+let all_ppx_runtime_deps_exn t ~dir lib_deps =
+  (* The [ppx_runtime_deps] of [Findlib.package] already holds the transitive closure. *)
+  let deep_traverse_externals = false in
```
Comment (rgrinberg) [inline src/lib_db.ml]: Is this flag set to `true` anywhere else? If not, then do we really need it?

Comment (diml) [reply src/lib_db.ml]: No, but I found that:
- splitting `all_ppx_runtime_deps_exn` in two functions made it easier to read
- making `fold_transitive_closure` not traverse external deps was a bit arbitrary, so making this function generic was clearer

**Pull request merged**